app/src/main/java/org/simple/clinic/sync/indicator/SyncIndicatorViewController.kt

```diff
@@ -129,4 +142,63 @@ class SyncIndicatorViewController @Inject constructor(
         }
         .flatMap { syncStream().mergeWith(errorsStream()) }
   }
+
+  private fun isSyncPending(events: Observable<UiEvent>): Observable<UiChange> {
+    val screenCreates = events.ofType<SyncIndicatorViewCreated>()
+
+    val isPatientSyncPending = screenCreates
+        .flatMap {
+          patientRepository
+              .pendingRecordsCount()
+              .map { count -> count > 0 }
+        }
+
+    val isBpSyncPending = screenCreates
+        .flatMap {
+          bloodPressureRepository
+              .pendingRecordsCount()
+              .map { count -> count > 0 }
+        }
+
+    val isPrescriptionSyncPending = screenCreates
+        .flatMap {
+          prescriptionRepository
+              .pendingRecordsCount()
+              .map { count -> count > 0 }
+        }
+
+    val isMedicalHistorySyncPending = screenCreates
+        .flatMap {
+          medicalHistoryRepository
+              .pendingRecordsCount()
+              .map { count -> count > 0 }
+        }
+
+    val isAppointmentSyncPending = screenCreates
+        .flatMap {
+          appointmentRepository
+              .pendingRecordsCount()
+              .map { count -> count > 0 }
+        }
+
+    val isCommunicationSyncPending = screenCreates
+        .flatMap {
+          communicationRepository
+              .pendingRecordsCount()
+              .map { count -> count > 0 }
+        }
+
+    return Observable
+        .mergeArray(
+            isPatientSyncPending,
+            isBpSyncPending,
+            isPrescriptionSyncPending,
+            isAppointmentSyncPending,
+            isMedicalHistorySyncPending,
+            isCommunicationSyncPending
+        )
+        .filter { isSyncPending -> isSyncPending }
+        .distinctUntilChanged()
+        .map { { ui: Ui -> ui.updateState(SyncPending) } }
+  }
```

Comment:
Unrelated to this PR, but I was looking at `startSync()` and I think we're starting a sync on every `SyncIndicatorViewClicked`, without unsubscribing from a previously started sync. We probably need to use `switchMap` there instead of `flatMap`. Can you check?